From abecf5804b45379fa7f3eae790d89e9c2d0917ca Mon Sep 17 00:00:00 2001
From: Noam Camus <ncamus@habana.ai>
Date: Mon, 08 Apr 2019 16:21:46 +0300
Subject: [PATCH] [SW-0] preboot: update regmap byte access

Do not access CS with byte even if it is 32bit aligned.
Behaviour is not defined.
Instead access in dword size as expected by CS and then refer to
relevant byte

Change-Id: I485eefd4d53041073153dabf428c36ac3620c2b6
---

diff --git a/common/preboot.h b/common/preboot.h
index 1951262..701661f 100644
--- a/common/preboot.h
+++ b/common/preboot.h
@@ -22,14 +22,14 @@
 
 static inline u8 regmap_readb(u64 offset)
 {
-	u8 reg = *((volatile u8 *)(HL_CFG_BASE + offset));
+	u32 reg = *((volatile u32 *)(HL_CFG_BASE + offset));
 
-	return reg;
+	return (u8)(reg & 0xFF);
 }
 
 static inline void regmap_writeb(u8 val, u64 offset)
 {
-	*((volatile u8 *)(HL_CFG_BASE + offset)) = val;
+	*((volatile u32 *)(HL_CFG_BASE + offset)) = (u32)val;
 }
 
 static inline u32 regmap_readl(u64 offset)
